File: sdk/python/examples/complete_workflow.py
"""
Example: Complete user workflow with blockchain integration
"""

import asyncio
from dchat import BlockchainClient, UserManager


async def main():
    # Initialize blockchain client
    blockchain = BlockchainClient.local()

    # Create user manager
    user_manager = UserManager(
        blockchain=blockchain,
        base_url="http://localhost:8080"
    )

    try:
        # Create two users
        print("Creating users...")
        alice = await user_manager.create_user("alice")
        bob = await user_manager.create_user("bob")

        print(f"Alice: {alice.user_id} (confirmed: {alice.on_chain_confirmed})")
        print(f"Bob: {bob.user_id} (confirmed: {bob.on_chain_confirmed})")

        # Send direct message
        print("\nSending direct message...")
        message = await user_manager.send_direct_message(
            sender_id=alice.user_id,
            recipient_id=bob.user_id,
            content="Hello Bob!",
        )
        print(f"Message sent: {message.message_id} (TX: {message.tx_id})")

        # Create channel
        print("\nCreating channel...")
        channel = await user_manager.create_channel(
            creator_id=alice.user_id,
            channel_name="General",
            description="General discussion",
        )
        print(f"Channel created: {channel.channel_id} (TX: {channel.tx_id})")

        # Post to channel
        print("\nPosting to channel...")
        post = await user_manager.post_to_channel(
            sender_id=alice.user_id,
            channel_id=channel.channel_id,
            content="Welcome everyone!",
        )
        print(f"Posted: {post.message_id} (TX: {post.tx_id})")

        # Check current block
        block_number = await blockchain.get_block_number()
        print(f"\nCurrent block: {block_number}")

    except Exception as e:
        print(f"Error: {e}")


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/rust/examples/complete_workflow.rs
// Example: Complete user workflow with blockchain integration

use dchat_sdk::blockchain::BlockchainClient;
use dchat_sdk::user::UserManager;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize blockchain client
    let blockchain = BlockchainClient::local();

    // Create user manager
    let user_manager = UserManager::new(blockchain, "http://localhost:8080".to_string());

    // Create two users
    println!("Creating users...");
    let alice = user_manager.create_user("alice").await?;
    let bob = user_manager.create_user("bob").await?;

    println!("Alice: {} (confirmed: {})", alice.user_id, alice.on_chain_confirmed);
    println!("Bob: {} (confirmed: {})", bob.user_id, bob.on_chain_confirmed);

    // Send direct message
    println!("\nSending direct message...");
    let message = user_manager
        .send_direct_message(&alice.user_id, &bob.user_id, "Hello Bob!", None)
        .await?;
    println!(
        "Message sent: {} (TX: {:?})",
        message.message_id, message.tx_id
    );

    // Create channel
    println!("\nCreating channel...");
    let channel = user_manager
        .create_channel(&alice.user_id, "General", Some("General discussion"))
        .await?;
    println!(
        "Channel created: {} (TX: {:?})",
        channel.channel_id, channel.tx_id
    );

    // Post to channel
    println!("\nPosting to channel...");
    let post = user_manager
        .post_to_channel(&alice.user_id, &channel.channel_id, "Welcome everyone!")
        .await?;
    println!("Posted: {} (TX: {:?})", post.message_id, post.tx_id);

    Ok(())
}
